lib/libvfio-user.c

```diff
@@ -1052,6 +1052,45 @@ handle_dirty_pages(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg)
     return ret;
 }
 
+static int
+handle_device_feature(vfu_ctx_t *vfu_ctx, vfu_msg_t *msg)
+{
+    struct vfio_user_device_feature *req = msg->in.iov.iov_base;
+
+    if (vfu_ctx->migration == NULL) {
+        return -EINVAL;
+    }
+
+    if (!migration_feature_supported(req->flags)) {
```

Comment:
Is it possible for msg->in.iov.iov_len to be smaller than the size of vfio_user_device_feature?

Reply:
Yes, is this the sort of thing we want an assertion for or do we want it to fail gracefully?

Reply:
I haven't checked that code, if it's something the client can control then it should be handled gracefully.